include/zephyr/bluetooth/conn.h

```diff
@@ -1614,6 +1616,8 @@ struct bt_conn_auth_info_cb {
  *  Register callbacks to handle authenticated pairing. Passing NULL
  *  unregisters a previous callbacks structure.
  *
+ *  @note Registering the same callback struct twice is not supported.
+ *
```

Comment:
This will actually return an error if you attempt to register multiple callbacks, regardless of whether they are the same